### PR TITLE
Remove clubs and projects's features

### PR DESCRIPTION
**Pull request by JJarrie**

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
(rebase needed)

File: src/SensioLabs/Connect/Api/Api.php [hunk 1]
```diff
@@ -67,50 +68,50 @@ public function getAccessToken()
...
         $url = $this->constructUrlWithAccessToken($url);
 
         $this->logger->info(sprintf('GET %s', $url));
 
-        return $this->processResponse($this->browser->get($url, array_merge($headers, $this->getAcceptHeader())));
+        return $this->processResponse($this->browser->sendRequest(new Request('GET', $url, array_merge($headers, $this->getAcceptHeader()))));
     }
 
-    public function submit($url, $method = 'POST', array $fields, $headers = array())
+    public function submit($url, $method = 'POST', array $fields = [], $headers = [])
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Api/Api.php]: please revert all unrelated VS changes - fixtures and XML files included

Comment (JJarrie) [reply src/SensioLabs/Connect/Api/Api.php]: They are reverted.

File: tests/fixtures/user.xml [hunk 2]
```diff
@@ -188,198 +188,17 @@
...
                                 <is-owner>false</is-owner>
                             </membership>
                                                                                                                                                                 <membership>
-                                <foaf:Group id="4c147ff8-2e0e-4fd5-b833-aa0cd5fa486f">
-    <foaf:name><![CDATA[SensioLabs]]></foaf:name>
-        <atom:link rel="https://rels.connect.sensiolabs.com/image" href="https://connect.sensiolabs.com/api/images/4c147ff8-2e0e-4fd5-b833-aa0cd5fa486f.png" type="image/png" />
-    <atom:link rel="self" href="https://connect.sensiolabs.com/api/clubs/4c147ff8-2e0e-4fd5-b833-aa0cd5fa486f" type="application/vnd.com.sensiolabs.connect+xml" />
-    <atom:link rel="self" href="https://connect.sensiolabs.com/api/alternates/4c147ff8-2e0e-4fd5-b833-aa0cd5fa486f" type="text/html" />
-    </foaf:Group>
+
```
Comment (nicolas-grekas) [inline tests/fixtures/user.xml]: can be removed :)

File: tests/SensioLabs/Connect/Tests/Security/Authentication/Token/ConnectTokenTest.php [hunk 3]
```diff
@@ -22,6 +22,9 @@
  */
 class ConnectTokenTest extends TestCase
 {
+    /**
+     * @group legacy
```
Comment (nicolas-grekas) [inline tests/SensioLabs/Connect/Tests/Security/Authentication/Token/ConnectTokenTest.php]: suspicious

File: tests/SensioLabs/Connect/Tests/Security/Authentication/Token/ConnectTokenTest.php [hunk 4]
```diff
@@ -32,6 +35,9 @@ public function testGetRolesWithStringUser()
         $this->assertEquals('ROLE_ADMIN', $roles[1]->getRole());
     }
 
+    /**
+     * @group legacy
```
Comment (nicolas-grekas) [inline tests/SensioLabs/Connect/Tests/Security/Authentication/Token/ConnectTokenTest.php]: suspicious

File: src/SensioLabs/Connect/Api/Entity/User.php [hunk 5]
```diff
@@ -40,7 +40,6 @@ protected function configure()
              ->addProperty('twitterUsername')
              ->addProperty('linkedInUrl')
              ->addProperty('facebookUid')
-             ->addProperty('projects')
              ->addProperty('memberships')
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Api/Entity/User.php]: to be removed?

File: src/SensioLabs/Connect/Api/Api.php [hunk 6]
```diff
@@ -77,7 +77,7 @@ public function get($url, $headers = [])
         return $this->processResponse($this->browser->sendRequest(new Request('GET', $url, array_merge($headers, $this->getAcceptHeader()))));
     }
 
-    public function submit($url, $method = 'POST', array $fields = [], $headers = [])
+    public function submit($url, $method = 'POST', array $fields, $headers = array())
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Api/Api.php]: to be reverted

File: src/SensioLabs/Connect/Api/Entity/User.php (hunk 5 above)
Comment (tucksaun) [reply src/SensioLabs/Connect/Api/Entity/User.php]: 👍

File: src/SensioLabs/Connect/Security/Authentication/Token/CompatibilitySerializationConnectTokenTrait.php [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace SensioLabs\Connect\Security\Authentication\Token;
+
+trait CompatibilitySerializationConnectTokenTrait
+{
+    public function __serialize()
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Security/Authentication/Token/CompatibilitySerializationConnectTokenTrait.php]: this should be serialize/unserialize, no leading `__`

File: src/SensioLabs/Connect/Security/Authentication/Token/AbstractConnectToken.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+if (method_exists(AbstractToken::class, '__serialize')) {
+    abstract class AbstractConnectToken extends AbstractToken
+    {
+        use SerializationConnectTokenTrait;
+    }
+} else {
+    abstract class AbstractConnectToken extends AbstractToken
+    {
+        use CompatibilitySerializationConnectTokenTrait;
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Security/Authentication/Token/AbstractConnectToken.php]: let's inline this one

File: src/SensioLabs/Connect/Security/Authentication/Token/ConnectToken.php [hunk 9]
```diff
@@ -53,7 +52,11 @@ public function getRoles()
             return $this->getUserRoles($user);
         }
 
-        return parent::getRoles();
+        if (method_exists(AbstractToken::class, 'getRoleNames')) {
+            return parent::getRoleNames();
+        } else {
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Security/Authentication/Token/ConnectToken.php]: let's remove all the "else" that are not needed (see below too)

File: src/SensioLabs/Connect/Security/Authentication/Token/ConnectToken.php [hunk 10]
```diff
@@ -53,6 +52,10 @@ public function getRoles()
             return $this->getUserRoles($user);
         }
 
+        if (method_exists(AbstractToken::class, 'getRoleNames')) {
```
Comment (nicolas-grekas) [inline src/SensioLabs/Connect/Security/Authentication/Token/ConnectToken.php]: all the BC layers should be moved to a PR on 4.3!

File: tests/SensioLabs/Connect/Tests/Api/ApiTest.php [hunk 11]
```diff
@@ -47,9 +47,6 @@ public function testAccessTokenAccessorsAndMutators()
         $this->assertNull($this->api->getAccessToken());
     }
 
-    /**
-     * @group legacy
```
Comment (nicolas-grekas) [inline tests/SensioLabs/Connect/Tests/Api/ApiTest.php]: all these should be removed in 4.3, they cannot be legit if the test case doesn't change

**Comment by nicolas-grekas** [comment]
Rebase + squash needed please.

**Comment by nicolas-grekas** [comment]
Merged as part of #74, thank you @JJarrie
